ee/clickhouse/queries/test/test_util.py

```diff
@@ -12,25 +13,41 @@
 from posthog.test.base import _create_event
 
 
-@freeze_time("2021-01-21")
 def test_get_earliest_timestamp(db, team):
-    _create_event(team=team, event="sign up", distinct_id="1", timestamp="2020-01-04T14:10:00Z")
-    _create_event(team=team, event="sign up", distinct_id="1", timestamp="2020-01-06T14:10:00Z")
+    with freeze_time("2021-01-21") as frozen_time:
+        _create_event(team=team, event="sign up", distinct_id="1", timestamp="2020-01-04T14:10:00Z")
+        _create_event(team=team, event="sign up", distinct_id="1", timestamp="2020-01-06T14:10:00Z")
 
-    assert get_earliest_timestamp(team.id) == datetime(2020, 1, 4, 14, 10, tzinfo=pytz.UTC)
+        assert get_earliest_timestamp(team.id) == datetime(2020, 1, 4, 14, 10, tzinfo=pytz.UTC)
 
-    _create_event(team=team, event="sign up", distinct_id="1", timestamp="1984-01-06T14:10:00Z")
-    _create_event(team=team, event="sign up", distinct_id="1", timestamp="2014-01-01T01:00:00Z")
-    _create_event(team=team, event="sign up", distinct_id="1", timestamp="2015-01-01T01:00:00Z")
+        frozen_time.tick(timedelta(seconds=1))
+        _create_event(team=team, event="sign up", distinct_id="1", timestamp="1984-01-06T14:10:00Z")
+        _create_event(team=team, event="sign up", distinct_id="1", timestamp="2014-01-01T01:00:00Z")
+        _create_event(team=team, event="sign up", distinct_id="1", timestamp="2015-01-01T01:00:00Z")
 
-    assert get_earliest_timestamp(team.id) == datetime(2015, 1, 1, 1, tzinfo=pytz.UTC)
+        assert get_earliest_timestamp(team.id) == datetime(2015, 1, 1, 1, tzinfo=pytz.UTC)
 
 
 @freeze_time("2021-01-21")
 def test_get_earliest_timestamp_with_no_events(db, team):
     assert get_earliest_timestamp(team.id) == datetime(2021, 1, 14, tzinfo=pytz.UTC)
 
 
+def test_get_earliest_timestamp_has_a_short_lived_cache(db, team):
+    with freeze_time("2021-01-21") as frozen_time:
+        with patch("posthog.queries.util.insight_sync_execute") as patched_sync_execute:
```

Comment:
Nit: This could be a decorator

Reply:
using the patch decorator here caused `RuntimeError: Database access not allowed, use the "django_db" mark, or the "db" or "transactional_db" fixtures to enable it`

Reply:
Ah. Python tests vs pytest strikes here in nasty ways.